ibis/pandas/execution/selection.py

```diff
@@ -307,5 +300,23 @@ def execute_selection_dataframe(op, data, scope=None, **kwargs):
         result = result.loc[predicate]
 
     if sort_keys:
-        result = util.compute_sorted_frame(sort_keys, result, **kwargs)
-    return result.reset_index(drop=True)
+        result, grouping_keys, ordering_keys = util.compute_sorted_frame(
+            result, order_by=sort_keys, **kwargs)
+    else:
+        grouping_keys = ordering_keys = ()
+
+    # return early if we have any temporary grouping or ordering columns
+    assert not grouping_keys, 'group by should never show up in Selection'
+    if not ordering_keys:
+        return result
+
+    # create a sequence of columns that we need to drop
```

Comment:
how do you track the temporary columns? (these are the uuid created ones right?)

Reply:
> how do you track the temporary columns?

Any column that is not in the original schema but is in the frame is deemed temporary.

> (these are the uuid created ones right?)
Yep